kiwi_lint/similar_string.py

```diff
@@ -0,0 +1,73 @@
+import string
+
+import astroid
+from pylint.checkers import BaseChecker, utils
+from pylint.interfaces import IAstroidChecker
+from textdistance import levenshtein
+
+
+class SimilarStringChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+    name = "similar-string-checker"
+
+    msgs = {
+        "R5011": (
+            '"%s" is similar to "%s". Try keeping them the same to reduce work for translators!',
+            "similar-string",
+            "Similar strings should be avoided for single translation",
+        )
+    }
+
+    _dict_of_strings = {}
+    threshold = 0.8
+
+    @staticmethod
+    def clean_string(text):
+        """
+        This method removes the operators and other punctuations
+        used in the string to avoid unwanted data to add up to
+        the similarity between the strings.
+        """
+        for char in text:
+            if char not in string.punctuation:
+                text += "" + char
+        return text
```

Comment:
This is still not returning appropriate results. Play with it and you will see.

Reply:
Oh, sorry! I didn't realize this.